checkov/common/runners/base_runner.py

```diff
@@ -122,14 +122,14 @@ def filter_ignored_paths(root_dir: str, names: List[Union[str, os.DirEntry]], ex
     # if excluded_paths = ['dir1/dir33', 'dir2/hello.yaml'], then we would scan dir1, but we would skip its subdirectories. We would scan
     # dir2 and its subdirectory, but we'd skip hello.yaml.
 
-    # first handle the legacy logic - this will also remove files starting with '.' but that's probably fine
+    # first handle the legacy logic 
     # mostly this will just remove those problematic directories hardcoded above.
     included_paths = included_paths or []
     for entry in list(names):
         path = entry if type(entry) == str else entry.name
         if path in ignored_directories:
             safe_remove(names, entry)
-        if path.startswith(".") and IGNORE_HIDDEN_DIRECTORY_ENV and path not in included_paths:
+        if path.startswith(".") and os.path.isdir(path) and IGNORE_HIDDEN_DIRECTORY_ENV and path not in included_paths:
```

Comment:
my guess is, that couple of tests fail, because of this change

Reply:
Interesting.  That was a change Barak added so that it didn't ignore dot files, only dot dirs.  We might need a Barak involved.

Reply:
fix using "included_paths" parameter